### PR TITLE
ROM README update

### PR DESCRIPTION
**Pull request by dylan-copeland**
The `README.md` file in the `rom` directory is updated with specific information about the ROM version of Laghos.

### CODE REVIEW

**Comment by kevinhkhuynh** [comment]
Do we want users to be able to use the main Laghos directories versions of metis and hypre? Or do we want to force them to use the same libraries as the paper? We currently use ParMetis 4.0.3 and Hypre 2.11.2 for our results.

**Comment by dylan-copeland** [comment]
@kevinhkhuynh I expect different versions of metis and hypre to have a small effect on results, on the level of round-off error (1e-12 or less). I don't see a reason to enforce a certain version.

**Comment by kevinhkhuynh** [comment]
> 
> 
> @kevinhkhuynh I expect different versions of metis and hypre to have a small effect on results, on the level of round-off error (1e-12 or less). I don't see a reason to enforce a certain version.

Sounds good to me. In that case, in another PR, I'll have the makefile look to ../../ for any libraries first, then dependencies second so users can use their main Laghos branch libraries that they've installed.

**Comment by chldkdtn** [comment]
@dylan-copeland didn't we get a very different result in our ROM when we do not use metis, for example, not ParMetis, in the past? Do you remember?

**Comment by dylan-copeland** [comment]
@chldkdtn This came up in the discussion about PR 119. The regression tests failed, because they compared basis values. The difference that metis vs. parmetis would make would be in the partitioning of the spatial domain in parallel, which should have round-off level differences but different DOF definitions. It is not possible to compare solution or basis values, with different DOF definitions and different numbers of DOFs on each process. However, the accuracy should have very little change.

**Comment by chldkdtn** [comment]
@dylan-copeland I was not talking about PR 119. Very beginning of this project when only you and I were working on it, I recall that if we use metis, not parmetis, we get very different solutions. It was definitely at the stage where our ROM for Laghos was premature, but it just came to my mind.

**Comment by dylan-copeland** [comment]
@chldkdtn I don't recall the details of that issue. Hopefully, we just saw a different DOF ordering. The important question is whether the difference is just in the DOF indices and parallel partitioning, or the solution accuracy. That is, we should compare L2 norms of differences (numerically integrated), not l2 norms (indices are reordered). This could be answered with the current state of the code by running our tests with metis and parmetis, to see if there is a difference in the L2 norm of the error (comparing FOM solutions and ROM errors). This is what the absolute regression tests do, so maybe they could be run with different builds to compare results.

**Comment by chldkdtn** [comment]
@dylan-copeland Thank you for editing "Code Structure" section. That looks good. Would you also edit Sections, "Purpose," "Characteristics?" Both sections are tailored for Laghos, not for Laghos ROM.

File: README.md [hunk 1]
```diff
@@ -201,31 +201,60 @@ download and building of hypre, METIS and MFEM.
...
 
-Some sample runs in 2D and 3D respectively are:
+A sample run of the offline stage is:
 ```sh
-mpirun -np 8 ./laghos -p 1 -dim 2 -rs 3 -tf 0.8 -pa
-mpirun -np 8 ./laghos -p 1 -dim 3 -rs 2 -tf 0.6 -pa -vis
+mpirun -np 8 ./laghos -p 1 -m data/cube01_hex.mesh -pt 211 -tf 0.8 -pa -offline -romsvds -romos -rostype interpolate -romsrhs -bef 1.0 -rpar 0
+mpirun -np 8 ./laghos -p 1 -m data/cube01_hex.mesh -pt 211 -tf 0.8 -pa -offline -romsvds -romos -rostype interpolate -romsrhs -bef 1.2 -rpar 1
+mpirun -np 8 ./laghos -p 1 -m data/cube01_hex.mesh -pt 211 -tf 0.8 -pa -offline -romsvds -romos -rostype interpolate -romsrhs -bef 0.8 -rpar 2
+./merge ./merge -nset 3 -ef 0.9999 -rhs -romos -rostype interpolate -nwinsamp 6
```
Comment (chldkdtn) [inline README.md]: ./merge is repeated twice

File: README.md [hunk 2]
```diff
@@ -201,31 +201,60 @@ download and building of hypre, METIS and MFEM.
 
 #### Sedov blast
 
-The main problem of interest for Laghos is the Sedov blast wave (`-p 1`) with
-partial assembly option (`-pa`).
+The 3D Sedov blast wave problem can be runned with `-p 1`.
 
-Some sample runs in 2D and 3D respectively are:
+A sample run of the offline stage is:
```
Comment (dylan-copeland) [inline README.md]: @siuwuncheung Should the ROM updates only be in `rom/README.md`, or is there a reason to update this main README as well? Note that Kevin and I have only been editing `rom/README.md`.

Comment (siuwuncheung) [reply README.md]: @dylan-copeland Thanks for your reminder. I have edited the wrong file.

File: rom/README.md [hunk 3]
```diff
@@ -1,59 +1,52 @@
...
+
+LaghosROM is based on the following article:
+
+> D. Copeland, K. Huynh, S.W. Cheung, and Y. Choi <br>
+> [Reduced order models for Lagrangian hydrodynamics]() <br>
+> *In Preparation*, 2021.
+
+To see the purpose of **Laghos**, please see README.md in Laghos directory.
 
 ## Characteristics
```
Comment (chldkdtn) [inline rom/README.md:38]: @dylan-copeland Thank you for editing "Characteristics" Section in rom/readme.md file. You can probably delete all the stuffs about Laghos and just leave the stuffs related to LaghosROM. You can also refer to readme.md file in laghos directory if someone wants to read about Laghos. I have done this for "Purpose" section for reference. Would you do the similar things for "Characteristics" Section also?

**Comment by chldkdtn** [comment]
@dylan-copeland @kevinhkhuynh @siuwuncheung I have introduced the term, "LaghosROM," to refer "the ROM version of Laghos." Let's use this terminology throughout this rom/README.md file.

**Comment by dylan-copeland** [comment]
@siuwuncheung The plot files added in the last commit dc77df0432b0579ccf79224ab5c4da2bbcb09ccc are somewhat large. By themselves, they are not very bad, but if we plan to put lots of plots in the README in the future it could make the Laghos repo large. Maybe we should consider whether plot files should be in the repo in general. @vladotomov @tzanio do you have some guidelines for us on limiting large files like these?

**Comment by vladotomov** [comment]
There aren't strict rules for Laghos, but these are good to remember:

- Add binary files only if they're completely necessary; try to keep them small.
- Files in any branch will be downloaded when the repo is cloned.
- Even if a file is deleted, it will be part of the history; do squash merges to remove unused files.

**Comment by chldkdtn** [comment]
@tzanio I guess that the copyright of CEED in README file of rom subdirectory must stay there although the rom work is not supported by CEED. What do you think, @tzanio ?

**Comment by tzanio** [comment]
>  I guess that the copyright of CEED in README file of rom subdirectory must stay there although the rom work is not supported by CEED

No, I think you should use a simpler README, like the one in the `serial/` directory

**Review by chldkdtn — COMMENTED** [review]
> There aren't strict rules for Laghos, but these are good to remember:
> 
>     * Add binary files only if they're completely necessary; try to keep them small.
> 
>     * Files in any branch will be downloaded when the repo is cloned.
> 
>     * Even if a file is deleted, it will be part of the history; do squash merges to remove unused files.



> > I guess that the copyright of CEED in README file of rom subdirectory must stay there although the rom work is not supported by CEED
> 
> No, I think you should use a simpler README, like the one in the `serial/` directory



> > I guess that the copyright of CEED in README file of rom subdirectory must stay there although the rom work is not supported by CEED
> 
> No, I think you should use a simpler README, like the one in the `serial/` directory



> > I guess that the copyright of CEED in README file of rom subdirectory must stay there although the rom work is not supported by CEED
> 
> No, I think you should use a simpler README, like the one in the `serial/` directory

@tzanio Not sure what you mean. I still see the same copyright statement of CEED in README file in the 'serial/' directory. 

> > I guess that the copyright of CEED in README file of rom subdirectory must stay there although the rom work is not supported by CEED
> 
> No, I think you should use a simpler README, like the one in the `serial/` directory

@tzanio  Not sure what you mean. I still see the same CEED copyright statement in README file in the 'serial/' directory.

**Comment by dylan-copeland** [comment]
It may be a better idea	to have plots and other presentation materials on a webpage, not in the Laghos repo. That way, we could keep the README file size small, with maybe a link to a webpage outside the repo, like mfem.org. The README could report error norms instead of showing plots, or even that could just go in an external webpage. Plots are nice for presentation, but I don't think they are very essential here, as the main point is that the error is small. Since results will change over time, and more results will be added in the future, it may not be very manageable to have it all contained in the Laghos repo.

In any case, since some plot files were changed in at least one commit, I suppose we should squash merge (whether we keep the plot files or not).

File: rom/README.md [hunk 4]
```diff
@@ -118,209 +86,112 @@ Other computational motives in Laghos include the following:
...
-- **CUDA** version in the [cuda/](./cuda/README.md) directory. This version supports GPU acceleration.
-- **RAJA** version in the [raja/](./raja/README.md) directory. This version supports GPU acceleration. See [GitHub](https://software.llnl.gov/RAJA/) for more information about RAJA.
-- **OCCA** version in the [occa/](./occa/README.md) directory. This version supports GPU and OpenMP acceleration. See the OCCA [website](http://libocca.org/) for more information.
-- **AMR** version in the [amr/](./amr/README.md) directory. This version supports dynamic adaptive mesh refinement.
-- **MFEM/engines**-based version in the
-  [engines-kernels](https://github.com/CEED/Laghos/tree/engines-kernels) branch.
+The results are generated by an MacBook Air (13-inch, 2017) 
+equipped with 1.8 GHz Dual-Core Intel Core i5. 
 
 ## Contact
```
Comment (chldkdtn) [inline rom/README.md:220]: We need to create rom@llnl.gov email list so that external people can ask questions to this email address. I will do that.

Comment (chldkdtn) [reply rom/README.md:220]: I have created surrogate@llnl.gov. ;)

**Review by chldkdtn — APPROVED** [review]

**Review by siuwuncheung — APPROVED** [review]

**Comment by chldkdtn** [comment]
Hi, @kevinhkhuynh, when you get a chance, would you review and approve?

**Review by kevinhkhuynh — APPROVED** [review]

**Pull request merged**